sql/pipelines/src/main/scala/org/apache/spark/sql/pipelines/graph/CoreDataflowNodeProcessor.scala

```diff
@@ -0,0 +1,232 @@
+          case flow: Flow => flow
+          case _ =>
+            throw new IllegalArgumentException(
+              s"Unsupported upstream node type for table ${table.displayName}: " +
+              s"${upstreamNodes.getClass}"
+            )
+        }
+        val resolvedFlowsToTable = flowsToTable.map { flow =>
+          resolvedFlowNodesMap.get(flow.identifier)
+        }
+
+        // Assign isStreamingTable (MV or ST) to the table based on the resolvedFlowsToTable
+        val tableWithType = table.copy(
+          isStreamingTableOpt = Option(resolvedFlowsToTable.exists(f => f.df.isStreaming))
+        )
+
+        // Table will be virtual in either of the following scenarios:
+        // 1. If table is present in context.fullRefreshTables
+        // 2. If table has any virtual inputs (flows or tables)
+        // 3. If the table pre-existing metadata is different from current metadata
+        val virtualTableInput = VirtualTableInput(
+          identifier = table.identifier,
+          specifiedSchema = table.specifiedSchema,
+          incomingFlowIdentifiers = flowsToTable.map(_.identifier).toSet,
+          availableFlows = resolvedFlowsToTable
+        )
+        resolvedInputs.put(table.identifier, virtualTableInput)
+        Seq(tableWithType)
+      case view: View =>
+        // For view, add the flow to resolvedInputs and return empty.
+        require(upstreamNodes.size == 1, "Found multiple flows to view")
+        upstreamNodes.head match {
+          case f: Flow =>
+            resolvedInputs.put(view.identifier, resolvedFlowNodesMap.get(f.destinationIdentifier))
+            Seq(view)
+          case _ =>
+            throw new IllegalArgumentException(
+              s"Unsupported upstream node type for view ${view.displayName}: " +
+              s"${upstreamNodes.getClass}"
+            )
+        }
+      case _ =>
+        throw new IllegalArgumentException(s"Unsupported node type: ${node.getClass}")
+    }
+  }
+}
+
+private class FlowResolver(rawGraph: DataflowGraph) extends Logging {
+
+  /** Helper used to track which confs were set by which flows. */
+  private case class FlowConf(key: String, value: String, flowIdentifier: TableIdentifier)
+
+  /** Attempts resolving a single flow using the map of resolved inputs. */
+  def attemptResolveFlow(
+      flowToResolve: UnresolvedFlow,
+      allInputs: Set[TableIdentifier],
+      availableResolvedInputs: Map[TableIdentifier, Input]): ResolvedFlow = {
+    val flowFunctionResult = flowToResolve.func.call(
+      allInputs = allInputs,
+      availableInputs = availableResolvedInputs.values.toList,
+      configuration = flowToResolve.sqlConf,
+      currentCatalog = flowToResolve.currentCatalog,
+      currentDatabase = flowToResolve.currentDatabase
+    )
+    val result =
+      flowFunctionResult match {
+        case f if f.dataFrame.isSuccess =>
+          // Merge the flow's inputs' confs into confs for this flow, throwing if any conflict
```

Comment:
Can you provide an example in the docs of why conflicting confs can't be supported?

Reply:
We actually don't throw if there's a conflict. changed this.